src/store/zoom.ts

```diff
@@ -0,0 +1,96 @@
+import { createSlice, PayloadAction, createSelector } from "@reduxjs/toolkit";
+
+interface ZoomState {
+  byId: {
+    [id: string]: {
+      id: string;
+      name: string;
+    };
+  };
+  allIds: string[];
+  online: string[];
+  hasSeenStart: boolean;
+  active: boolean;
+};
+
+const initialState: ZoomState = {
+  byId: {},
+  allIds: [],
+  online: [],
+  hasSeenStart: false,
+  active: false
+};
+
+type UserEventAction = PayloadAction<{ id: string, name: string }>;
+
+const { reducer, actions, name } = createSlice({
+  name: "zoomUsers",
+  reducers: {
+    callStarted: (state) => {
+      state.active = true;
+      state.online = [];
+    },
+    callEnded: (state) => {
+      state.active = false;
+      state.hasSeenStart = true;
+      state.online = [];
+    },
+    userLeft: (state, { payload }: UserEventAction) => {
+      const knownUser = state.byId.hasOwnProperty(payload.id);
+      if (!knownUser) {
+        state.byId[payload.id] = payload;
```

Comment:
I would run this line irrespective of if there's already an entry for the user (ie. delete lines 39, 40, 42, 43, and 51). That way, if someone updates their Zoom name, the update is propagated.